### PR TITLE
TSC minutes, 9th April

### PR DESCRIPTION
**Pull request by Idclip**
Signed-off-by: Nick Avramoussis <4256455+Idclip@users.noreply.github.com>

### CODE REVIEW

File: tsc/meetings/2020-04-09.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+VDB methods. Using the unit tests would provide a base set of examples to start
+writing documentation. Nick, concern that as the unit tests have not necessarily
+been designed for this that some of the logic in there could be "inaccurate" in
+terms of an optimal implementation solution. Additional care to be taken that
+the unit tests are first and foremost that, with clear documentation of the
+intended example. Nick to create a ticket with a task to identity some of the
+larger tools missing documentation and corresponding unit tests which could be
+used as a first doc attempt.
+
+4b) Hessian computation of VDB grids
```
Comment (Idclip) [inline tsc/meetings/2020-04-09.md:62]: https://jira.aswf.io/browse/OVDB-141

File: tsc/meetings/2020-04-09.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+of various applications against matrices of otherwise incompatible sizes and to
+better correspond to a default value of colour (alpha) or homogeneous coordiante
+(w). The alternative would be to explicitly error on this behaviour. AX
+currently already errors on assignment of containers of mismatching sizes.
+
+4) Nick proposed that every TSC meeting additionally begin with a quick recap
+of newly opened forum posts or github issues. Unanimous consensus. Two topics
+for discussion.
+
+4a) OpenVDB Documentation
```
Comment (Idclip) [inline tsc/meetings/2020-04-09.md:47]: https://jira.aswf.io/browse/OVDB-142

File: tsc/meetings/2020-04-09.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+from a scalar Grid. Currently unsupported. Ken, should be fairly trivial to
+implement. However, as the result per value is a matrix, there is no native way
+to store this information as VDB does not register matrix grids by default.
+Registering these grid types would be one option - however compile and runtime
+variants of "strided" grids (where each voxel/tile value contains an array of
+scalar values of a fixed size) were discussed as alternative solutions which
+could add more flexibility. Some potential confusion when inferring values on
+a grid with an "intrinsic" type i.e. Vec3f vs float[3] collocated vs staggered
+interpretation. General agreement that runtime variants would be interesting to
+pursue. Nick to create a ticket and reply to issue.
```
Comment (danrbailey) [inline tsc/meetings/2020-04-09.md:74]: Intriguing, I wonder what the performance implications would be to switch to a runtime equivalent to Vec3f, it sounds like quite a lot of engineering to answer that question. I suppose that could follow the proposal to move implementation out of the hierarchy classes (tree, grid, etc) into dedicated tools to simplify this task, but it's probably still non-trivial. It sounds like the idea of introducing compile-time Mat3/Mat4 variants was discarded?

I'll add the runtime strided grids to the roadmap as well, as it's worth discussing in that context.

Comment (jmlait) [reply tsc/meetings/2020-04-09.md:74]: Not discarded, it was raised.   But then naturally you ask about [6] and other odd-ball sized grids.  Adding four new template expansions (float/double and mat3/mat4) to destroy compile time vs solving it with a runtime tuple solution?

Comment (danrbailey) [reply tsc/meetings/2020-04-09.md:74]: Personally, I would prioritize work aimed at reducing compile times generally such as moving implementation out of grid/tree classes as mentioned and introducing explicit template instantiation or some other compile-time registration mechanism, then consider this runtime grid proposal in that new light. If it need not destroy compile times, perhaps that changes how we would make the decision to proceed here.

**Pull request merged**